content/5_Plat/Convex_Hull.mdx

```diff
@@ -267,7 +267,192 @@ int main()
 
 ### Solution
 
-<IncompleteSection />
+With the Monotone Chain algorithm, we start by sorting the given $n$ points in ascending order with respect to their $x$ coordinates. If two points have the same $x$ value, then we will look at the $y$ coordinate. 
+
+Next, we will calculate the convex hull in two parts - the upper and the lower hull. Firstly, we observe that the starting and ending points of both upper and lower hulls are the same. They are the points with the lowest and highest $x$ value, $P_0$ and $P_{n-1}$. We primarily add $P_0$ and $P_1$ to the hull. Note that $P_1$ doesn't necessarily have to be a point of the convex hull at the end. Then, starting from $P_2$, we iterate through the sorted points and add them to the hull. Let's denote the current point being inspected as $P_{k}$ and the last point in the hull as $P_{i}$. While adding new points, we want to make sure that there is no right turn among all segments of the hull, just like in the Graham Algorithm discussed above. To achieve this, the segment $P_{i-1}P_i$ should always be on the left side of the segment $P_{i-1}P_k$. This can be calculated by a cross product trick: 
```

Comment:
```suggestion
Next, we will calculate the convex hull in two parts - the upper and the lower hull. Firstly, we observe that the starting and ending points of both upper and lower hulls are the same. They are the points with the lowest and highest $x$ value respectively, $P_0$ and $P_{n-1}$. We start by adding $P_0$ and $P_1$ to the hull. (Note that $P_1$ doesn't necessarily have to be on the convex hull at the end). Then, starting from $P_2$, we iterate through the sorted points and add them to the hull. Let's denote the current point being added as $P_{k}$ and the last point still on the hull as $P_{i}$. When adding new points, we want to make sure that there is no right turn among all segments of the hull, just like in the Graham Scan algorithm discussed above. To achieve this, the segment $P_{i-1}P_i$ should always be on the left side of the segment $P_{i-1}P_k$. This can be calculated by using a cross-product: 
```